### DISCUSSION TITLE
How to clean the langfuse cache data periodically?

### DISCUSSION THREAD

**Discussion by He-Huang** [langfuse/langfuse#2211] [Support] [answered]
Langfuse is a great tool for diagnosing LLM application. It is storing the traced data in PostgreSQL database. However, the size of the history data grows quickly. In my case, for a single LLM project, the db size reached 8GB within two weeks. Therefore, what's the best practice to periodically clean the history cache data? I wonder whether there is a configurable parameter in langfuse or I need to create a SQL clean task additionally myself.

Thank you very much.

**Comment by marcklingen** [comment, marked as answer]
Currently you need to run clean up tasks on the Postgres db yourself when self-hosting. You can purge the trace and observation tables based on the created_at timestamp.

Data retention is a planned feature for Langfuse v3.
